tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1966,6 +1937,327 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
+					replicasLarge = 5
+					replicasSmall = 3
+				case v1alpha1.TiKVMemberType:
+					replicasLarge = 4
+					replicasSmall = 3
+				case v1alpha1.TiDBMemberType:
+					replicasLarge = 3
+					replicasSmall = 2
+				}
+				ginkgo.It(fmt.Sprintf("should work for %s", comp), func() {
+					ginkgo.By("Deploy initial tc")
+					tc := fixture.GetTidbCluster(ns, fmt.Sprintf("scale-out-scale-in-%s", strings.ToLower(string(comp))), utilimage.TiDBV4)
+					switch comp {
+					case v1alpha1.PDMemberType:
+						tc.Spec.PD.Replicas = replicasLarge
+					case v1alpha1.TiKVMemberType:
+						tc.Spec.TiKV.Replicas = replicasLarge
+					case v1alpha1.TiDBMemberType:
+						tc.Spec.TiDB.Replicas = replicasLarge
+					}
+					utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc, 5*time.Minute, 10*time.Second)
+
+					ginkgo.By(fmt.Sprintf("Scale in %s", comp))
+					err := controller.GuaranteedUpdate(genericCli, tc, func() error {
+						switch comp {
+						case v1alpha1.PDMemberType:
+							tc.Spec.PD.Replicas = replicasSmall
+						case v1alpha1.TiKVMemberType:
+							tc.Spec.TiKV.Replicas = replicasSmall
+						case v1alpha1.TiDBMemberType:
+							tc.Spec.TiDB.Replicas = replicasSmall
+						}
+						return nil
+					})
+					framework.ExpectNoError(err, "failed to scale in %s for TidbCluster %s/%s", comp, ns, tc.Name)
+					ginkgo.By(fmt.Sprintf("Wait for %s to be in ScalePhase", comp))
+					switch comp {
+					case v1alpha1.PDMemberType:
+						utiltc.MustWaitForComponentPhase(cli, tc, v1alpha1.PDMemberType, v1alpha1.ScalePhase, 3*time.Minute, 10*time.Second)
+					case v1alpha1.TiKVMemberType:
+						utiltc.MustWaitForComponentPhase(cli, tc, v1alpha1.TiKVMemberType, v1alpha1.ScalePhase, 3*time.Minute, 10*time.Second)
+					case v1alpha1.TiDBMemberType:
+						utiltc.MustWaitForComponentPhase(cli, tc, v1alpha1.TiDBMemberType, v1alpha1.ScalePhase, 3*time.Minute, 10*time.Second)
+					}
+					log.Logf(fmt.Sprintf("%s is in ScalePhase", comp))
+					ginkgo.By("Wait for tc ready")
+					err = oa.WaitForTidbClusterReady(tc, 3*time.Minute, 10*time.Second)
+					framework.ExpectNoError(err, "failed to wait for TidbCluster %s/%s ready after scale in %s", ns, tc.Name, comp)
+					log.Logf("tc is ready")
+
+					pvcUIDs := make(map[string]string)
+					ginkgo.By("Check PVC annotation tidb.pingcap.com/pvc-defer-deleting")
+					err = wait.Poll(10*time.Second, 3*time.Minute, func() (done bool, err error) {
+						for ordinal := replicasSmall; ordinal < replicasLarge; ordinal++ {
+							var pvcSelector labels.Selector
+							switch comp {
+							case v1alpha1.PDMemberType:
+								pvcSelector, err = member.GetPVCSelectorForPod(tc, v1alpha1.PDMemberType, int32(ordinal))
+							case v1alpha1.TiKVMemberType:
+								pvcSelector, err = member.GetPVCSelectorForPod(tc, v1alpha1.TiKVMemberType, int32(ordinal))
+							case v1alpha1.TiDBMemberType:
+								pvcSelector, err = member.GetPVCSelectorForPod(tc, v1alpha1.TiDBMemberType, int32(ordinal))
+							}
```

Comment:
Can we just use `pvcSelector, err = member.GetPVCSelectorForPod(tc, comp, int32(ordinal))` here?

if so, ditto for L2066~2072.